src/static/riot/analytics/analytics.tag

```diff
@@ -0,0 +1,540 @@
+                start.setDate(start.getDate() - 6)
+                self.start_date = {
+
+                    year: start.getFullYear(),
+                    month: start.getMonth() + 1,
+                    day: start.getDate(),
+                }
+            } else if (unit_selection === 'year') {
+                self.start_date = {
+                    year: new Date().getFullYear(),
+                    month: 0,
+                    day: 1,
+                }
+            }
+
+            self.update_analytics(self.start_date, self.end_date, 'day', false)
+
+            if (unit_selection != 'year') {
+                $(self.refs.day_units).click()
+            } else {
+                $(self.refs.month_units).click()
+            }
+        }
+
+        self.show_date_selectors = function (e) {
+            $(self.refs.date_selection_container).removeClass('hidden')
+        }
+
+        // Chart Units (Months, Weeks, Days)
+        self.update_chart_resolution = function(e) {
+            let id = e.target.id
+            var unit_selection = id.split('-')[0]
+
+            if( unit_selection === 'day') {
+                self.time_unit = 'day'
+            } else if (unit_selection === 'week') {
+                self.time_unit = 'week'
+            } else {
+                self.time_unit = 'month'
+            }
+
+
+            self.competitionsChart.options.scales.xAxes[0].time.unit = unit_selection
+            self.submissionsChart.options.scales.xAxes[0].time.unit = unit_selection
+            self.usersChart.options.scales.xAxes[0].time.unit = unit_selection
+            self.competitionsChart.update()
+            self.submissionsChart.update()
+            self.usersChart.update()
+
+            self.update_analytics(self.start_date, self.end_date, self.time_unit, false)
+        }
+
+        self.download = function (filename, text) {
+            var element = document.createElement('a');
+            element.setAttribute('href', 'data:text/plain;charset=utf-8,' + encodeURIComponent(text));
+            element.setAttribute('download', filename);
+
+            element.style.display = 'none';
+            document.body.appendChild(element);
+
+            element.click();
+
+            document.body.removeChild(element);
+        }
+
+        self.format_data_to_csv = function (data) {
+            let all_text = ''
+            _.forOwn(data, function(value, key) {
+                all_text += key + '\n'
+                if (Array.isArray(value)) {
+                    all_text += array_to_text(value)
+                } else {
+                    all_text += value
+                }
+                all_text += '\n'
+            })
+            return all_text
+        }
+
+        function array_to_text(arr) {
+            x_string = ""
+            y_string = ""
+            _(arr).forEach(function (element) {
+                x_string += element.x
+                x_string += ', '
```

Comment:
can do this all at once w/ a formatted string
```javascript
x_string += `${element.x}, `
```

Reply:
Nice. I hadn't seen js formatted strings until now.